### PR TITLE
[T3A3][T15-A4]AngShiYa

### PR DESCRIPTION
**Pull request by AngShiYa**

### CODE REVIEW

**Comment by AngShiYa** [comment]
Ready for review

File: src/seedu/addressbook/data/person/Address.java [hunk 1]
```diff
@@ -8,12 +8,26 @@
...
+    public static final String MESSAGE_ADDRESS_CONSTRAINTS = 
+            "Person addresses should be in the format block, street, unit, postal code";
+    public static final String ADDRESS_VALIDATION_REGEX = ".+, .+, .+, .+";
+    
+    public static final int ADDRESS_INDEX_BLOCK = 0;
+    public static final int ADDRESS_INDEX_STREET = 1;
+    public static final int ADDRESS_INDEX_UNIT = 2;
+    public static final int ADDRESS_INDEX_POSTALCODE = 3;
 
     public final String value;
```
Comment (nishantbudhdev) [inline src/seedu/addressbook/data/person/Address.java]: No need for `value` any more.

File: src/seedu/addressbook/data/person/Block.java [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+    public static final String MESSAGE_BLOCK_CONSTRAINTS = 
+            "Person block should be either number only or number followed by an alphabet";
+    public static final String BLOCK_VALIDATION_REGEX = "[\\d]*[\\w]?";
+    
+    private String block;
+    
+    public Block(String block) throws IllegalValueException {
+        if (!isBlockValid(block)) {
+            throw new IllegalValueException(MESSAGE_BLOCK_CONSTRAINTS);
+        }
```
Comment (nishantbudhdev) [inline src/seedu/addressbook/data/person/Block.java]: Great job on using regex within each part correctly.

**Comment by nishantbudhdev** [comment]
@AngShiYa stellar work. 

* As mentioned you don't need to use `value` anymore, hence you need to replace all instances/ calls to it.
* You've managed to modify nearly all the required files.

Please close the PR after reading comments.

**Pull request closed without merging**